spec/components/layout_for_admin_spec.rb

```diff
@@ -16,4 +16,25 @@ def component_name
 
     assert_select 'meta[name="robots"][content="noindex,nofollow,noimageindex"]', visible: false
   end
+
+  it "can receive a custom js filename" do
+    allow_any_instance_of(ActionView::Base).to receive(:javascript_include_tag)
+    expect_any_instance_of(ActionView::Base).to receive(:javascript_include_tag).with("admin").once
+
+    render_component(
+      browser_title: "Hello, admin page",
+      environment: "production",
+      js_filename: "admin",
+    )
+  end
+
+  it "can receive a custom css filename" do
+    expect_any_instance_of(ActionView::Base).to receive(:stylesheet_link_tag).with("admin", media: "all")
```

Comment:
Interested to know how these tests work - I was expecting an `assert` or similar, how do they work?

Reply:
I've wrote a lengthy (sorry about that) comment [here](https://github.com/alphagov/govuk_publishing_components/pull/3993#discussion_r1579178525) which hopefully addresses this.

Let me know if any of it doesn't make sense